### PR TITLE
11.0 mig account chart update

### PR DESCRIPTION
**Pull request by amcor**
cc @guadaltech

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@amcor are you going to continue with the rest of the missing features?

**Comment by amcor** [comment]
Hi @pedrobaeza , the module is working with all the old features (account, account.tax and fiscal.position) but ... 
I have to test it with more details to include more fields in the updates, specially in account.tax, with account and fiscal.position i dont see any problem
I dont know if there is another feature to have in count.

**Comment by pedrobaeza** [comment]
Yes, I mean extra fields like account or tax tags.

**Comment by sbidoul** [comment]
@pedrobaeza about tax tags, do you have a recommended strategy to set tags on taxes after an openupgrade migration?

**Comment by pedrobaeza** [comment]
If we have this working, I would recommend to pass this chart update over the migrated database for "synchronizing" data in the company and in the template. What do you think?

**Comment by sbidoul** [comment]
@pedrobaeza yes, that's a possibility. Does not work for manually created taxes though.

I see the old `base_code_id` etc... fields on account.tax now point to account_tax_group. Do you think we could exploit that information somehow?

**Comment by pedrobaeza** [comment]
Yes, OpenUpgrade already converts v8 tax codes into tax groups (https://github.com/OCA/OpenUpgrade/blob/4987b871e0e03192a96dcd9afd54d83165af6297/addons/account/migrations/9.0.1.1/pre-migration.py#L111), and parent taxes into tags. If you think that another distribution can be done, we can change OpenUpgrade scripts.

**Comment by pedrobaeza** [comment]
As information, I always avoid manually created taxes, and if needed, I add in the company's custom module that taxes definition in the chart template. That way, updating this on newer version and updating chart template, I get everything correct.

**Comment by sbidoul** [comment]
Note Odoo has utility function to update tax tags: 

https://github.com/odoo/odoo/blob/cf1ec5d3167b304923c91291cf13d1c8be73c290/addons/account/models/chart_template.py#L24

**Review by sbidoul — APPROVED** [review]
This works, I tested. The problems I had are not related to the migration iteself.

**Comment by sbidoul** [comment]
Note it does not work with child taxes. But the problem is present in 10 too.

**Comment by pedrobaeza** [comment]
#593 is supposed to fix it for v10.

File: account_chart_update/wizard/wizard_chart_update.py [hunk 1]
```diff
@@ -0,0 +1,783 @@
...
+            else:
+                for key, value in self.diff_fields(template, tax).items():
+                    tax[key] = value
+                _logger.debug(_("Updated tax %s."), template.name)
+            wiz_tax.update_tax_id = tax
+
+    def _create_account_from_template(self, account_template):
+        return self.env["account.account"].create({
+            'name': account_template.name,
+            'currency_id': account_template.currency_id,
```
Comment (pedrobaeza) [inline account_chart_update/wizard/wizard_chart_update.py]: It lacks final `.id` (see #683)

**Comment by pedrobaeza** [comment]
I have made a major refactor in #693. This will need to be redone after merging it, but this was not working correctly. I will do it any way.

**Review by pedrobaeza — APPROVED** [review]
I have brought the improvements from v10 and adapt it to v11, with full coverage, so it's enough for merging!

Thanks to all.

**Pull request merged**